src/main/java/org/wildfly/swarm/plugin/AbstractFractionsMojo.java

```diff
@@ -49,58 +49,27 @@
-    public List<MavenProject> fractionProjects() {
-        return this.project.getDependencyManagement().getDependencies()
-                .stream()
-                .filter(this::isSwarmProject)
-                .map(this::toProject)
-                .filter(e -> e != null)
-                .filter(FractionDetector::isFraction)
-                .collect(Collectors.toList());
-    }
-
-    public synchronized  Map<String,Fraction> fractions() {
-        if ( CACHE != null ) {
-            return CACHE;
-        }
-        List<MavenProject> fractionProjects = fractionProjects();
-
-        Map<String, Fraction> fractions = new TreeMap<>();
-        fractionProjects.forEach(d -> fractions.put(d.getGroupId() + ":" + d.getArtifactId(),
-                new Fraction(d.getGroupId(), d.getArtifactId(), d.getVersion())));
-
-        Fraction container = new Fraction("org.wildfly.swarm", "container", this.project.getVersion());
-
-        fractions.put("org.wildfly.swarm:container", container);
+    private static List<MavenProject> PROBABLY_FRACTIONS = null;
 
-        fractionProjects.forEach(fractionProject -> {
-            final Fraction current = fractions.get(fractionProject.getGroupId() + ":" + fractionProject.getArtifactId());
+    public List<MavenProject> probablyFractionProjects() {
+        if ( PROBABLY_FRACTIONS == null ) {
 
-            current.setName(fractionProject.getName());
-            current.setDescription(fractionProject.getDescription());
-            Properties properties = fractionProject.getProperties();
-            current.setTags(properties.getProperty(FRACTION_TAGS_PROPERTY_NAME, ""));
-            current.setInternal(Boolean.parseBoolean(properties.getProperty(FRACTION_INTERNAL_PROPERTY_NAME)));
-            current.setStabilityIndex(StabilityLevel.of(fractionProject));
-            Set<Artifact> deps = fractionProject.getArtifacts();
-
-            for (Artifact each : deps) {
-                Fraction f = fractions.get(each.getGroupId() + ":" + each.getArtifactId());
-                if (f == null) {
-                    continue;
-                }
-                if (f.getGroupId().equals("org.wildfly.swarm") && f.getArtifactId().equals("bootstrap")) {
-                    continue;
-                }
-                current.addDependency(f);
-            }
-            current.addDependency(container);
-        });
-
-        CACHE = fractions;
+            PROBABLY_FRACTIONS = this.project.getDependencyManagement().getDependencies()
+                    .stream()
+                    .filter(this::isSwarmProject)
+                    .map(this::toProject)
+                    .collect(Collectors.toList());
+        }
 
-        return fractions;
+        return PROBABLY_FRACTIONS;
+    }
 
+    public synchronized  Set<FractionMetadata> fractions() {
+        return probablyFractionProjects()
+                .stream()
+                .map(FractionRegistry.INSTANCE::of)
+                .filter(e->e!=null)
```

Comment:
Ah thank you! I knew there had to be a predicate I could just reference but
my JDK knowledge isn't complete enough wrt Java8.

On Monday, August 29, 2016, George Gastaldi <notifications@github.com>
wrote:

> In src/main/java/org/wildfly/swarm/plugin/AbstractFractionsMojo.java
> <https://github.com/wildfly-swarm/wildfly-swarm-fraction-plugin/pull/12#discussion_r76720727>
> :
>
> >
> > +    public synchronized  Set<FractionMetadata> fractions() {
> > +        return probablyFractionProjects()
> > +                .stream()
> > +                .map(FractionRegistry.INSTANCE::of)
> > +                .filter(e->e!=null)
>
> Objects::nonNull could be used here
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wildfly-swarm/wildfly-swarm-fraction-plugin/pull/12/files/929556f2cad4822a49ea3f9a0dbc7d5bb5f45b5e#r76720727>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAA-T8Eq7gQkdfm1WLP9lYC6DDc1Ir4Aks5qk5CWgaJpZM4Jv8yV>
> .
>